cpp/order/Steinhardt.cc

```diff
@@ -0,0 +1,495 @@
+			}
+			else
+			{
+				m_Wli = std::shared_ptr<complex<float> >(new complex<float>[Np], std::default_delete<complex<float>[]>());
+			}
+		}
+		else
+		{
+			if (m_average && m_norm)
+			{
+				m_QliAveNorm = std::shared_ptr<float>(new float[Np], std::default_delete<float[]>());
+			}
+			else if (m_norm)
+			{
+				m_QliNorm = std::shared_ptr<float>(new float[Np], std::default_delete<float[]>());
+			}
+		}
+	}	
+
+void Steinhardt::compute(const locality::NeighborList *nlist, const vec3<float> *points, unsigned int Np)
+{
+	// Conditional reinitialize arrays if size differs from previous call.
+	if (m_Np != Np)
+	{
+		m_Np = Np;
+		Steinhardt::reallocateArrays(Np);
+	}
+	// Computes the base Q required for each specialized order parameter
+	Steinhardt::baseCompute(nlist, points);
+	
+	if (m_average){
+		Steinhardt::computeAve(nlist, points);
+	}
+	if (m_Wl){
+		if (m_average && m_norm){
+			Steinhardt::computeAveNormWl();
+		}
+		else if (m_norm){
+			Steinhardt::computeNormWl();
+		}
+		else if (m_average){
+			Steinhardt::computeAveWl();
+		}
+		else {
+			Steinhardt::computeWl();
+		}
+	}
+	else {
+		if (m_average && m_norm){
+			Steinhardt::computeAveNorm();
+		}
+		else if (m_norm){
+			Steinhardt::computeNorm();
+		}
+	}
+}
+
+void Steinhardt::baseCompute(const locality::NeighborList *nlist, const vec3<float> *points)
+    {
+    nlist->validate(m_Np, m_Np);
+
+    memset((void*) m_Qlmi.get(), 0, sizeof(complex<float>)*(2*m_l+1)*m_Np);
+    memset((void*) m_Qli.get(), 0, sizeof(float)*m_Np);
+
+    parallel_for(tbb::blocked_range<size_t>(0, m_Np),
+        [=] (const blocked_range<size_t>& r)
+        {
+	// May not be necessary check
+        const float rminsq = m_rmin * m_rmin;
+        const float rmaxsq = m_rmax * m_rmax;
+        const float normalizationfactor = 4*M_PI/(2*m_l+1);
+        const size_t *neighbor_list(nlist->getNeighbors());
+	// --------------------------
+
+        bool Qlm_exists;
+        m_Qlm_local.local(Qlm_exists);
+        if (!Qlm_exists)
+            {
+            m_Qlm_local.local() = new complex<float> [2*m_l+1];
+            memset((void*) m_Qlm_local.local(), 0, sizeof(complex<float>) * (2*m_l+1));
+            }
+
+        size_t bond(nlist->find_first_index(r.begin()));
+        // for each reference point
+        for (size_t i = r.begin(); i != r.end(); i++)
+            {
+            unsigned int neighborcount(0);
+            const vec3<float> ref(points[i]);
+            for (; bond < nlist->getNumBonds() && neighbor_list[2*bond] == i; ++bond)
+                {
+                const unsigned int j(neighbor_list[2*bond + 1]);
+
+                if (i == j)
+                    {
+                    continue;
+                    }
+
+                const vec3<float> delta = m_box.wrap(points[j] - ref);
+                const float rsq = dot(delta, delta);
+
+		// Replace conditional with function call above.
```

Comment:
Is this your comment?

Reply:
No that I think was left over from the hackathon, but I must have missed it